lib/vector-core/src/metrics/mod.rs

```diff
@@ -109,25 +109,25 @@ pub fn get_controller() -> crate::Result<&'static Controller> {
 
 /// Take a snapshot of all gathered metrics and expose them as metric
 /// [`Event`]s.
-pub fn capture_metrics(controller: &Controller) -> impl Iterator<Item = Event> {
-    let mut events: Vec<Event> = Vec::new();
+pub fn capture_metrics(controller: &Controller) -> impl Iterator<Item = Metric> {
```

Comment:
Made this method more specific to avoid needing to downcast into `Metric` when manipulating the `Event`.